src/Nethermind/Nethermind.Abi/AbiType.TypeMapping.cs

```diff
@@ -44,19 +48,45 @@ protected static AbiType GetForCSharpType(Type type)
             }
         }
 
+        protected static bool IsMappingRegistered<T>()
+        {
+            return _typeMappings.ContainsKey(typeof(T));
+        }
+
         protected static void RegisterMapping<T>(AbiType abiType)
         {
-            _typeMappings[typeof(T)] = abiType;
+            lock (_registerLock)
+            {
+                Dictionary<Type, AbiType> typeMappings = new(_typeMappings)
+                {
+                    [typeof(T)] = abiType
+                };
+
+                _typeMappings = typeMappings;
```

Comment:
A concurrent reader might be reading from the `_typeMappings` instance.

Reply:
Adds are very infrequent (basically non-existent) so rather than having to pay for a heavier data structure like ConcurrentDictionary or a lock on every read we can use an unprotected raw Dictionary if we replace the pointer entirely on add